pkg/watch/health.go

```diff
@@ -85,44 +101,48 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 // service it is monitoring. Every HEALTHCHECK_INTERVAL it
 // performs a health check and writes that information to
 // consul
-func (p *PodWatch) MonitorHealth(node string, store kp.Store, shutdownCh chan bool) {
+func (p *PodWatch) MonitorHealth(store kp.Store, statusChecker StatusCheck, shutdownCh chan bool) {
 	for {
 		select {
 		case <-time.After(HEALTHCHECK_INTERVAL):
-			p.checkHealth(node, p.manifest.StatusPort, store)
+			p.checkHealth(store, statusChecker)
 		case <-shutdownCh:
 			return
 		}
 	}
 }
 
-func (p *PodWatch) checkHealth(node string, port int, store kp.Store) {
-	resp, err := kp.HttpStatusCheck(node, port)
+func (p *PodWatch) checkHealth(store kp.Store, sc StatusCheck) {
+	resp, err := sc.Check()
 	health, err := resultFromCheck(resp, err)
```

Comment:
Not that you want to be composing these functions, at least not here. To present the caller with a simple, hard-to-misuse interface, this should probably be folded into `Check()`. It does make sense to keep it as a separate function, so you can unit test it without faking/mocking HTTP calls.